box-content-sdk/src/main/java/com/box/androidsdk/content/requests/BoxRequestsMetadata.java

```diff
@@ -0,0 +1,221 @@
+         *
+         * @param map    values of the item to add metadata to.
+         * @return  request with the updated values.
+         */
+        protected AddFileMetadata setValues(Map<String,Object> map) {
+            mBodyMap.putAll(map);
+            return this;
+        }
+    }
+
+    /**
+     * Request for getting metadata on a file
+     */
+    public static class GetFileMetadata extends BoxRequest<BoxMetadata, GetFileMetadata> {
+        /**
+         * Creates a get file metadata request with the default parameters
+         *
+         * @param requestUrl    URL of the file metadata endpoint
+         * @param session       the authenticated session that will be used to make the request with
+         */
+        public GetFileMetadata(String requestUrl, BoxSession session) {
+            super(BoxMetadata.class, requestUrl, session);
+            mRequestMethod = Methods.GET;
+        }
+    }
+
+    /**
+     * Request for udpating metadata on a file
+     */
+    public static class UpdateFileMetadata extends BoxRequest<BoxMetadata, UpdateFileMetadata> {
+
+        /**
+         * ENUM that defines all possible update operations.
+         */
+        public enum Operations {
+            ADD("add"),
+            REPLACE("replace"),
+            REMOVE("remove"),
+            TEST("test");
+
+            private String mName;
+
+            private Operations(String name) {
+                mName = name;
+            }
+
+            @Override
+            public String toString() { return mName; }
+        }
+
+        private BoxArray<BoxMetadataUpdateTask> mUpdateTasks;
+
+        /**
+         * Creates a update file metadata request with the default parameters
+         *
+         * @param requestUrl    URL of the file metadata endpoint
+         * @param session       the authenticated session that will be used to make the request with
+         */
+        public UpdateFileMetadata(String requestUrl, BoxSession session) {
+            super(BoxMetadata.class, requestUrl, session);
+            mRequestMethod = Methods.PUT;
+            mContentType = ContentTypes.JSON_PATCH;
+            mUpdateTasks = new BoxArray<BoxMetadataUpdateTask>();
+        }
+
+        /**
+         * Updates the values of the item used in the request.
+         *
+         * @param updateTasks    task list for metadata update.
+         * @return  request with the updated values.
+         */
+        protected UpdateFileMetadata setUpdateTasks(BoxArray<BoxMetadataUpdateTask> updateTasks) {
+            mBodyMap.put(BoxRequest.JSON_OBJECT, updateTasks);
+            return this;
+        }
+
+        /**
+         * Add new update task to request.
+         * @param operation The operation to apply.
+         * @param key The key.
+         * @param value The value for the path (key). Can leave blank if performing REMOVE operation.
+         */
+        public void addUpdateTask(Operations operation, String key, String value) {
```

Comment:
have this return UpdateFileMetadata instead of void so developers can use the builder pattern like: 
```java
updateRequest.addUpdateTask(op, k, v)
    .addUpdateTask(op2, k2, v2);
```